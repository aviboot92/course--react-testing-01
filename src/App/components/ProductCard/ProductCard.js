import React from 'react'
import PropTypes from 'prop-types'
import { Pane, Button } from 'evergreen-ui'

const ProductCard = ({ name, age, imgSrc, description, onCtaClick }) => {
	return (
		<Pane
			borderWidth="1px"
			borderColor="#EEE"
			borderStyle="solid"
			borderRadius="5px"
			display="grid"
			gridTemplateColumns="1fr 1fr"
		>
			<img alt={name} src={imgSrc} width="100%" />
			<Pane padding="10px">
				<h3 style={{ marginTop: 0 }}>
					{name}, {age}
				</h3>
				<h5>{description}</h5>
				<Button onClick={() => onCtaClick({ imgSrc, name })}>Adopt</Button>
			</Pane>
		</Pane>
	)
}

ProductCard.propTypes = {
	age: PropTypes.number.isRequired,
	description: PropTypes.string.isRequired,
	onCtaClick: PropTypes.func.isRequired,
	imgSrc: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
}

export default ProductCard
