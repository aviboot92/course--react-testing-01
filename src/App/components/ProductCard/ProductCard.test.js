import React from 'react'
import { shallow } from 'enzyme'
import { Button } from 'evergreen-ui'
import ProductCard from './ProductCard'

const essentialProps = {
  age: 5,
  description: 'Affectionate, Cheeky',
  onCtaClick: jest.fn(),
  imgSrc: 'imgSrc',
  name: 'Buba',
}

describe('<ProductCard />', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ProductCard {...essentialProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render essential elements', () => {
    const wrapper = shallow(<ProductCard {...essentialProps} />)
    expect(wrapper.find('img').props()).toEqual({
      alt: 'Buba',
      src: 'imgSrc',
      width: '100%',
    })
    expect(wrapper.find('h3').text()).toBe('Buba, 5')
    expect(wrapper.find('h5').text()).toBe('Affectionate, Cheeky')
    expect(wrapper.find(Button).childAt(0).text()).toBe('Adopt')
  })
  it('should call onCtaClick with imgSrc and name when button is clicked', () => {
    const wrapper = shallow(<ProductCard {...essentialProps} />)
    essentialProps.onCtaClick.mockClear()
    wrapper.find(Button).simulate('click')
    expect(essentialProps.onCtaClick).toHaveBeenCalledWith({
      imgSrc: essentialProps.imgSrc,
      name: essentialProps.name
    })
  })
})
