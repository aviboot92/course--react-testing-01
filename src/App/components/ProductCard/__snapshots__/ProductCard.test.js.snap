// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render as expected 1`] = `
<withTheme(Pane)
  borderColor="#EEE"
  borderRadius="5px"
  borderStyle="solid"
  borderWidth="1px"
  display="grid"
  gridTemplateColumns="1fr 1fr"
>
  <img
    alt="Buba"
    src="imgSrc"
    width="100%"
  />
  <withTheme(Pane)
    padding="10px"
  >
    <h3
      style={
        Object {
          "marginTop": 0,
        }
      }
    >
      Buba
      , 
      5
    </h3>
    <h5>
      Affectionate, Cheeky
    </h5>
    <withTheme(Button)
      onClick={[Function]}
    >
      Adopt
    </withTheme(Button)>
  </withTheme(Pane)>
</withTheme(Pane)>
`;
